pkg/component/gardener/apiserver/apiserver.go

```diff
@@ -181,6 +181,12 @@ func (g *gardenerAPIServer) Deploy(ctx context.Context) error {
 	if err := managedresources.CreateForSeedWithLabels(ctx, g.client, g.namespace, ManagedResourceNameRuntime, false, managedResourceLabels, runtimeResources); err != nil {
 		return err
 	}
+	// We need to wait for the gardener-apiserve service to get created by the gardener-apiserver-runtime MR
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+	if err := managedresources.WaitUntilHealthy(timeoutCtx, g.client, g.namespace, ManagedResourceNameRuntime); err != nil {
+		return err
+	}
```

Comment:
Do we really need to wait for the entire MR to become healthy? If we are only interested in the `Service`, perhaps we could retry reading it for half a minute or so, and fail otherwise?